version-compatibility/src/fuel_tx.rs

```diff
@@ -25,3 +25,32 @@ fn release_0_58_2_can_deserialize_latest() {
     // Then
     let _ = latest_tx.expect("Deserialization failed");
 }
+
+#[test]
+fn latest_can_deserialize_previous_tx_pointer() {
+    for idx in 0..u16::MAX {
```

Comment:
```suggestion
    for idx in 0..=u16::MAX {
```

Would it work if we used `u16::MAX`?